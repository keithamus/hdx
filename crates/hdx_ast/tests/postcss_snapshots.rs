mod helpers;

#[test]
#[cfg(feature = "serde")]
fn postcss_media() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/apply.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_atrule_brackets() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/at-rule-brackets.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_atrule_decls() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/atrule-decls.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_atrule_empty() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/atrule-empty.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_atrule_no_params() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/atrule-no-params.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_atrule_no_semicolon() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/atrule-no-semicolon.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_atrule_params() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/atrule-params.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_atrule_rules() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/atrule-rules.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_between() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/between.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_colon_selector() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/colon-selector.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_comments() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/comments.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_custom_properties() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/custom-properties.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_decls() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/decls.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_empty() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/empty.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_escape() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/escape.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_extends() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/extends.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_function() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/function.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_ie_prodid() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/ie-progid.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_important() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/important.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_inside() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/inside.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_no_selector() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/no-selector.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_prop() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/prop.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_quotes() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/quotes.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_raw_decl() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/raw-decl.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_rule_at() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/rule-at.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_rule_no_semicolon() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/rule-no-semicolon.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_selector() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/selector.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_semicolons() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/semicolons.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_tab() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/tab.css");
}
