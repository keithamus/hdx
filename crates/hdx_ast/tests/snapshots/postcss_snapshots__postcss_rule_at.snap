---
source: crates/hdx_ast/tests/postcss_snapshots.rs
expression: result.output.unwrap()
---
{
  "type": "stylesheet",
  "rules": [
    {
      "type": "stylerule",
      "selectors": [
        {
          "components": [
            {
              "type": "tag",
              "value": {
                "Html": {
                  "A": {
                    "kind": "Ident",
                    "offset": 0,
                    "len": 1
                  }
                }
              }
            }
          ],
          "comma": null
        }
      ],
      "type": "style-declaration",
      "open": {
        "kind": "LeftCurly",
        "offset": 2,
        "len": 1
      },
      "declarations": [
        {
          "type": "property",
          "name": {
            "kind": "Ident",
            "offset": 8,
            "len": 5
          },
          "colon": {
            "kind": "Colon",
            "offset": 13,
            "len": 1
          },
          "value": {
            "LengthPercentage": {
              "type": "px",
              "value": {
                "kind": "Dimension",
                "len": 4,
                "unit": "px"
              }
            }
          },
          "important": null,
          "semicolon": {
            "kind": "Semicolon",
            "offset": 19,
            "len": 1
          }
        },
        {
          "type": "property",
          "name": {
            "kind": "Ident",
            "offset": 25,
            "len": 6
          },
          "colon": {
            "kind": "Colon",
            "offset": 31,
            "len": 1
          },
          "value": {
            "values": [
              {
                "kind": "Whitespace",
                "offset": 32,
                "len": 1
              },
              {
                "kind": "Number",
                "offset": 33,
                "len": 1
              },
              {
                "kind": "Whitespace",
                "offset": 34,
                "len": 1
              },
              {
                "kind": "AtKeyword",
                "offset": 35,
                "len": 6
              }
            ]
          },
          "important": null,
          "semicolon": {
            "kind": "Semicolon",
            "offset": 41,
            "len": 1
          }
        }
      ],
      "rules": [
        {
          "type": "AtRule",
          "name": {
            "kind": "AtKeyword",
            "offset": 47,
            "len": 6
          },
          "prelude": {
            "values": [
              {
                "kind": "Whitespace",
                "offset": 53,
                "len": 1
              },
              {
                "kind": "Ident",
                "offset": 54,
                "len": 6
              }
            ]
          },
          "block": {
            "type": "Block",
            "start": 60,
            "open_curly": {
              "kind": "LeftCurly",
              "offset": 61,
              "len": 1
            },
            "declarations": [
              {
                "type": "Declaration",
                "name": {
                  "kind": "Ident",
                  "offset": 71,
                  "len": 6
                },
                "colon": {
                  "kind": "Colon",
                  "offset": 77,
                  "len": 1
                },
                "value": {
                  "values": [
                    {
                      "kind": "Whitespace",
                      "offset": 78,
                      "len": 1
                    },
                    {
                      "kind": "Dimension",
                      "offset": 79,
                      "len": 5,
                      "unit": "px"
                    }
                  ]
                },
                "important": null,
                "semicolon": {
                  "kind": "Semicolon",
                  "offset": 84,
                  "len": 1
                }
              }
            ],
            "rules": [],
            "close_curly": {
              "kind": "RightCurly",
              "offset": 90,
              "len": 1
            }
          }
        }
      ],
      "close": {
        "kind": "RightCurly",
        "offset": 92,
        "len": 1
      }
    }
  ]
}
